criu/page-xfer.c

```diff
@@ -206,7 +227,7 @@ static int open_page_server_xfer(struct page_xfer *xfer, int fd_type, unsigned l
 	/* Push the command NOW */
 	tcp_nodelay(xfer->sk, true);
 
-	if (read(xfer->sk, &has_parent, 1) != 1) {
+	if (__recv(xfer->sk, &has_parent, 1, 0) != 1) {
 		pr_perror("The page server doesn't answer");
```

Comment:
will tls_recv set a proper errno wich is reported by pr_perror?

Reply:
the value for the `errno` variable will be set by `recv()` which is called by `gnutls_record_recv()` (in the `tls_recv()` function).

For example, by terminating the `criu dump` process in the middle of a post-copy migration, on the destination side, I'm getting the following error message:
```
Error (criu/tls.c:128): tls: Failed receiving data: Error in the pull function.
Error (criu/page-xfer.c:1236): Error reading async data from page server: Connection reset by peer 
```

However, `gnutls_record_recv()` may also fail with a different error, such as `GNUTLS_E_INVALID_SESSION`, `GNUTLS_E_INTERNAL_ERROR`, etc. In this case the value of `errno` will not be set.

Reply:
I mean that pr_perror always prints errno, but __recv can call tls_recv which doesn't set errno and the error message will contain errno which has nothing with a described error...

Reply:
I changed tls_send() and tls_recv() to set `errno = EIO` in case of error.